src/pci_virtio_net_vpnkit.c

```diff
@@ -78,15 +78,37 @@
 #include <xhyve/mevent.h>
 #include <xhyve/virtio.h>
 
-#include "protocol.h"
-#include "utils.h"
-
 #define WPRINTF(format, ...) printf(format, __VA_ARGS__)
 
 #define VTNET_RINGSZ 1024
 #define VTNET_MAXSEGS 32
 
 /*
+ * wire protocol
+ */
+struct msg_init {
+	uint8_t magic[5]; /* VMN3T */
+	uint32_t version;
+	uint8_t commit[40];
+} __packed;
+
+#define CMD_ETHERNET 1
+struct msg_common {
+	uint8_t command;
+} __packed;
+
+struct msg_ethernet {
+	uint8_t command; /* CMD_ETHERNET */
+	char uuid[36];
+} __packed;
+
+struct vif_info {
+	uint16_t mtu;
+	uint16_t max_packet_size;
+	uint8_t mac[6];
+} /*__packed*/;
```

Comment:
Clang complained (with an error) that the struct was already naturally packed and so therefore the annotation was unnecessary. I left it in mainly for its documentary effects (personally I think the error was actively harmful...)